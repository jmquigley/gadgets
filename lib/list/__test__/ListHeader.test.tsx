'use strict';

import * as assert from 'assert';
import {shallow} from 'enzyme';
import * as React from 'react';
import {getDefaultListHeaderProps, ListHeader} from '../index';

test('Test retrieval of ListHeader props object', () => {
	const props = getDefaultListHeaderProps();

	assert(props);
	expect(props).toMatchSnapshot();
});

test('Test the creation of a ListHeader control with simple title', () => {
	const ctl = shallow(
		<ListHeader
			title="test title"
		/>
	);

	assert(ctl);
	expect(ctl).toMatchSnapshot();
});

// TODO: test case for validating props object creator
// TODO: disabling the accordion item
// TODO: make the accordion item invisible
