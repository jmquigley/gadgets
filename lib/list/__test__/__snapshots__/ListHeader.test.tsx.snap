// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test retrieval of ListHeader props object 1`] = `
Object {
  "className": "",
  "contentEditable": false,
  "direction": "right",
  "disabled": false,
  "errorMessage": "",
  "focus": false,
  "height": "",
  "hiddenLeftButton": false,
  "hiddenRightButton": false,
  "href": Object {
    "sizing": "normal",
  },
  "id": "",
  "layout": 5,
  "left": "",
  "leftButton": null,
  "location": "none",
  "noedit": false,
  "nohover": true,
  "noripple": false,
  "obj": "ListHeader",
  "onBlur": [Function],
  "onChange": [Function],
  "onClick": [Function],
  "onDoubleClick": [Function],
  "onFocus": [Function],
  "onKeyDown": [Function],
  "onKeyPress": [Function],
  "onMouseOut": [Function],
  "onUpdate": [Function],
  "padding": "",
  "rightButton": null,
  "selected": false,
  "sizing": "normal",
  "stacked": false,
  "style": Object {},
  "testing": true,
  "theme": null,
  "title": "â€‹",
  "tooltip": "",
  "top": "",
  "useedit": false,
  "visible": true,
  "widget": null,
  "width": "",
  "xcss": null,
}
`;

exports[`Test the creation of a ListHeader control with simple title 1`] = `
<ThemeProvider
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "color": "#000000",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.5s",
    }
  }
>
  <Item
    className="ui-list-header nohover"
    contentEditable={false}
    direction="right"
    disabled={false}
    errorMessage=""
    focus={false}
    height=""
    hiddenLeftButton={false}
    hiddenRightButton={false}
    href={
      Object {
        "sizing": "normal",
      }
    }
    id=""
    layout={5}
    left=""
    leftButton={null}
    location="none"
    noedit={false}
    nohover={true}
    noripple={false}
    obj="ListHeader"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onDoubleClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onMouseOut={[Function]}
    onUpdate={[Function]}
    padding=""
    rightButton={null}
    selected={false}
    sizing="normal"
    stacked={false}
    style={Object {}}
    testing={true}
    theme={null}
    title="test title"
    tooltip=""
    top=""
    useedit={false}
    visible={true}
    widget={null}
    width=""
    xcss={null}
  />
</ThemeProvider>
`;
